File: objc-demo/ChatDemo-UI3.0/Code/Controllers/EMLaunchViewController.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */

#import "EMLaunchViewController.h"

@interface EMLaunchViewController () <EMClientDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *launchImageView;

@end

@implementation EMLaunchViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [self setBackgroundColor];
    [self setLaunchAnimation];
    
    BOOL isAutoLogin = [EMClient sharedClient].isAutoLogin;
    if (isAutoLogin){
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.65 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            [[NSNotificationCenter defaultCenter] postNotificationName:KNOTIFICATION_LOGINCHANGE object:@YES];
        });
    } else {
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.65 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            [[NSNotificationCenter defaultCenter] postNotificationName:KNOTIFICATION_LOGINCHANGE object:@NO];
        });
    }
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    [UIApplication sharedApplication].statusBarStyle = UIStatusBarStyleLightContent;
}

- (void)setBackgroundColor
{
    CAGradientLayer *gradient = [CAGradientLayer layer];
    gradient.frame = [UIScreen mainScreen].bounds;
    gradient.colors = [NSArray arrayWithObjects:(id)LaunchTopColor.CGColor,(id)LaunchBottomColor.CGColor,nil];
    [gradient setStartPoint:CGPointMake(0.0, 0.0)];
    [gradient setEndPoint:CGPointMake(0.0, 1.0)];
    [self.view.layer insertSublayer:gradient atIndex:0];
}

- (void)setLaunchAnimation
{
    _launchImageView.animationImages = @[[UIImage imageNamed:@"logo1"],
                                         [UIImage imageNamed:@"logo2"],
                                         [UIImage imageNamed:@"logo3"],
                                         [UIImage imageNamed:@"logo4"],
                                         [UIImage imageNamed:@"logo5"],
                                         [UIImage imageNamed:@"logo6"],
                                         [UIImage imageNamed:@"logo7"],
                                         [UIImage imageNamed:@"logo8"],
                                         [UIImage imageNamed:@"logo9"],
                                         [UIImage imageNamed:@"logo10"],
                                         [UIImage imageNamed:@"logo11"]];
    _launchImageView.animationDuration = 1.65;
    _launchImageView.animationRepeatCount = 1;
    [_launchImageView startAnimating];
    
    _launchImageView.top = (KScreenHeight - _launchImageView.height)/2;
    _launchImageView.left = (KScreenWidth - _launchImageView.width)/2;
}


@end

File: swift-demo/Hyphenate-Demo-Swift/Code/Controllers/EMLaunchViewController.swift
//
//  EMLaunchViewController.swift
//  Hyphenate-Demo-Swift
//
//  Created by dujiepeng on 2017/6/12.
//  Copyright © 2017 dujiepeng. All rights reserved.
//

import UIKit
import Hyphenate


let animationDuration = 1.65 

class EMLaunchViewController: UIViewController, EMClientDelegate {
    
    @IBOutlet weak var launchImageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setBackgroundColor() 
        setLauchAnimation() 

        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + animationDuration, execute: {
            if EMClient.shared().isLoggedIn {
                NotificationCenter.default.post(name: NSNotification.Name(rawValue:KNOTIFICATION_LOGINCHANGE), object: NSNumber(value: true)) 
            } else {
                NotificationCenter.default.post(name: NSNotification.Name(rawValue:KNOTIFICATION_LOGINCHANGE), object: NSNumber(value: false)) 
            }
        }) 
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated) 
        UIApplication.shared.statusBarStyle = UIStatusBarStyle.lightContent 
    }
    
    func setBackgroundColor () {
        let gradient = CAGradientLayer.init() 
        gradient.frame = UIScreen.main.bounds 
        gradient.colors =  [LaunchTopColor.cgColor, LaunchBottomColor.cgColor] 
        gradient.startPoint = CGPoint.init(x: 0.0, y: 0.0) 
        gradient.endPoint = CGPoint.init(x: 0.0, y: 1.0) 
        view.layer.insertSublayer(gradient, at: 0) 
    }
    
    func setLauchAnimation () {
        launchImageView.animationImages = [
            UIImage(named: "logo1")!,
            UIImage(named: "logo2")!,
            UIImage(named: "logo3")!,
            UIImage(named: "logo4")!,
            UIImage(named: "logo5")!,
            UIImage(named: "logo6")!,
            UIImage(named: "logo7")!,
            UIImage(named: "logo8")!,
            UIImage(named: "logo9")!,
            UIImage(named: "logo10")!,
            UIImage(named: "logo11")!
        ] 
        
        launchImageView.animationDuration = animationDuration 
        launchImageView.animationRepeatCount = 1 
        launchImageView.startAnimating() 
        
        launchImageView.top(top: (kScreenHeight - launchImageView.height()) / 2) 
        launchImageView.left(left: (kScreenWidth - launchImageView.width()) / 2) 
    }
}
